airflow/api_connexion/endpoints/variable_endpoint.py

```diff
@@ -14,41 +14,76 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import Optional
 
-# TODO(mik-laj): We have to implement it.
-#     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8142
+from flask import Response, make_response, request
+from marshmallow import ValidationError
 
+from airflow.api_connexion.exceptions import BadRequest, NotFound
+from airflow.api_connexion.schemas.variable_schema import variable_collection_schema, variable_schema
+from airflow.models import Variable
+from airflow.utils.session import provide_session
 
-def delete_variable():
+
+def delete_variable(variable_key: str) -> Response:
     """
     Delete variable
     """
-    raise NotImplementedError("Not implemented yet.")
+    Variable.delete(variable_key)
```

Comment:
I suggest we check the existence of this key and raise NotFound exception if key is not found.
What do you think?

Reply:
I initially had this in the implementation, but removed it later after noticing openapi spec did not specify a 404 response for this endpoint. I think it's better to return 404 when the key does not exist. @mik-laj thoughts on adding 404 to the openapi spec?

Reply:
I agree. That would be helpful.